aten/src/ATen/native/cuda/MultinomialKernel.cu

```diff
@@ -326,7 +327,7 @@ void multinomial_with_replacement_kernel_impl(
     int maxThreads = props->maxThreadsPerBlock;
     int maxShared = props->sharedMemPerBlock;
     int requiredShared = (numCategories < maxThreads ? numCategories : maxThreads)
-                         * (sizeof(scalar_t) + sizeof(accscalar_t));
+                         * (sizeof(accscalar_t) + 1);
```

Comment:
why do you need +1 here?

Reply:
Ah, I got it wrong. It was supposed to be in the line above.
Anyway, I need that because `numCategories` may be 1. If that's the case, I need at least one more element in the shared memory.
By the way, I guess when `numCategories` equals 1, I can just short-circuit and copy `self` to `result`.